drake/systems/framework/value.h

```diff
@@ -148,5 +157,18 @@ class VectorValue : public Value<VectorBase<T>*> {
   std::unique_ptr<VectorBase<T>> owned_value_;
 };
 
+// The following explicit instantiations are in libDrakeSystemFramework.
+// It is necessary to explicitly instantiate, and export, any implementation
+// of AbstractValue that would otherwise be implicitly instantiated elsewhere
+// in the library. If you do not create and export an explicit instantiation,
+// the Mach-O linker will generate a hidden typeinfo symbol in
+// libDrakeSystemFramework, and another hidden symbol in other libraries. These
+// typeinfos will have identical contents, but will not be identical for
+// the purpose of dynamic_cast, which will thwart unerasure of AbstractValue.
+
+
+extern template class DRAKESYSTEMFRAMEWORK_EXPORT Value<VectorBase<double>*>;
+extern template class DRAKESYSTEMFRAMEWORK_EXPORT VectorValue<double>;
```

Comment:
I *think* this will fail to compile on Windows when building the library with the message:
`'__declspec(dllexport)' and 'extern' are incompatible on an explicit instantiation`. The export macro should go with the instantiation rather than the extern declaration.